graphql-apollo-server-tutorial.md

```diff
@@ -533,19 +527,19 @@ const resolvers = {
 };
 {{< /highlight >}}
 
-The context should be the same for all resolvers now. Every resolver who needs to access the context, or in this case the `me` user, can do so by using the third argument of the resolver function.
+The context should be the same for all resolvers now. Every resolver that needs to access the context, or in this case the `me` user, can do so using the third argument of the resolver function.
 
-The fourth argument in a resolver function, the info argument, isn't used very often, because it only gives you internal information about the GraphQL request. It can be used for debugging, error handling, and advanced monitoring and tracking. You don't need to worry about it for now.
+The fourth argument in a resolver function, the info argument, isn't used very often, because it only gives you internal information about the GraphQL request. It can be used for debugging, error handling, advanced monitoring, and tracking. You don't need to worry about it for now.
 
-On the side, a couple of words about the a resolver's return values: As you have witnessed, a resolver can return arrays, objects and scalar types, but it has to be defined in the matching type definitions too. For instance, the type definition has to define an array or non-nullable field in order to have the resolvers working appropriately. What about JavaScript promises? Often you will make a request to a data source (database, RESTful API) in a resolver and thus return a JavaScript promise in the resolver. GraphQL can deal with it and waits for the promise being resolved. Only then the result is mapped to the type definitions. That's why you don't need to worry about asynchronous requests to your data source later on.
+A couple of words about the a resolver's return values: a resolver can return arrays, objects and scalar types, but it has to be defined in the matching type definitions. The type definition has to define an array or non-nullable field to have the resolvers working appropriately. What about JavaScript promises? Often, you will make a request to a data source (database, RESTful API) in a resolver, returning a JavaScript promise in the resolver. GraphQL can deal with it, and waits for the promise to resolve, except the result is mapped to the type definitions. That's why you don't need to worry about asynchronous requests to your data source later.
```

Comment:
"waits for the promise to resolve, except the result is mapped to the type definitions." -> "waits for the promise to resolve."